woocommerce/woocommerce-android
# The payment link option in the menu on order details should be removed

Closes: #6668


### Description

Removes redundant "share payment link" action from the order details screen, because now "collect button" opens payment method selection screen which includes this action 

### Testing instructions

* Create an order in the "ducks-in-row" branch
* Notice that "share payment" action in the order details screen is available
* Create an order in the current branch
* Notice that "share payment" action in the order details screen is not available

### Images/gif


https://user-images.githubusercontent.com/4923871/174961634-dc6271d8-498b-4fdf-a0be-353e073c6b27.mp4
